resource_cache/resource_cache.go

```diff
@@ -451,7 +451,36 @@ func (o *ObjectCache) Status(resourceIdent ResourceIdent, object client.Object)
 // update field on the internal resource. If the update is true, then the object will by applied, if
 // it is false, then the object will be created.
 func (o *ObjectCache) ApplyAll() error {
+	first := map[ResourceIdent]map[types.NamespacedName]*k8sResource{}
+	last := map[ResourceIdent]map[types.NamespacedName]*k8sResource{}
 	for k, v := range o.data {
+		gvk, err := utils.GetKindFromObj(o.scheme, k.GetType())
+		if err != nil {
+			return err
+		}
+		kind := gvk.Kind
+		if kind == "Deployment" || kind == "Job" || kind == "CronJob" {
```

Comment:
Could we recast this as a lookup rather than a conditional? Would make it more extensible should we decide to add more buckets later and reduces the cognitive load of reading a multi-part conditional.

Something like:
```
bucketMap := map[string]map[ResourceIdent]map[types.NamespacedName]*k8sResource{}
bucketMap["Deployment"] = last
bucketMap["Job"] = last
bucketMap["CronJob"] = last

toBucket = first
if mapBucket, ok := bucketMap[kind]; ok {
    toBucket = mapBucket
}

toBucket[k] = v
```